### ISSUE TITLE
com.apple.AuthenticationServices.AuthorizationError error 1000. Caused by Debug Mode

### ISSUE THREAD

**Issue by jeroendelau** [invertase/react-native-apple-authentication#127]
When I run the app On Device, in debug mode I get an error.
If I run the app On the Simulator, not in debug mode, I actually get to use Apple Sign In.

Could it be that not on device, but debug mode that is causing: 
com.apple.AuthenticationServices.AuthorizationError error 1000.

**Comment by mikehardy** [comment]
Hi there! It's not caused by debug mode. It's something wrong with the project config?
What is it? Good question, this is sort of a catch-all error from Apple

https://github.com/invertase/react-native-apple-authentication/#troubleshouting
And https://github.com/invertase/react-native-apple-authentication/issues?q=is%3Aissue+1000

If you determine what the solution is (try the ones listed in those issues), you could add to the troubleshooting section with a quick PR to the README

**Comment by jeroendelau** [comment]
Hmm, interesting indeed. I'll do my best to describe what I found in detail. If others find the same I'd be happy to issue a pull request for the docs. Else, it might just be me.

- First tried built from `cli.js run-ios "--simulator=iPhone 8"` -- ERROR
- Tried build in x-code -> device -> build configuration = debug --ERROR
- Tried build in x-code -> device -> build configuration = release -- works!
- Tried build in x-code -> simulator -> build configuration = release -- NO ERROR 

The two build configs are 100% identical except for the flag debug-release.
When it says ERROR, it is immediately upon clicking the button. 
In the simulator, I saw signs of life but I did not go through the whole process.

But I'll dig a bit deeper and see if I can pinpoint it a bit better.

**Comment by augustosamame** [comment]
I was getting this error: com.apple.AuthenticationServices.AuthorizationError error 1000 because I had happily skipped the part in the docs where I had to Add Apple Sign In capabilities in Xcode and Apple Developer. Double check you have done every step here:  https://github.com/invertase/react-native-apple-authentication/blob/master/docs/INITIAL_SETUP.md

**Comment by DanPHP7** [comment]
I got the same error, I've been enable the Capability "Sign In With Apple" on xcode, I made the certificate and the key with the service "Sign In With Apple" enabled, and got the same f***ing error, some onde can help me ?

```jsx
async appleAuthRequestResponse() {

      const response = await appleAuth.performRequest({
        requestedOperation: AppleAuthRequestOperation.LOGIN,
        requestedScopes: [
          AppleAuthRequestScope.EMAIL,
          AppleAuthRequestScope.FULL_NAME,
        ],
      });
      const { user, email } = response;
      console.tron.log('user', user);
      console.tron.log('email', email);
}
```

**Comment by mikehardy** [comment]
There are quite a few possible solutions, scan the repo especially the most recently updated, there are good tips in the other ones, we're looking for someone to add a tips-n-tricks section to the README that consolidates them https://github.com/invertase/react-native-apple-authentication/issues?q=is%3Aissue+1000+sort%3Aupdated-desc

**Comment by DanPHP7** [comment]
I did it
TARGETS>Signing & Capabilities>+Capability then search sign in then add it.
But the problem persist

**Comment by DanPHP7** [comment]
I don't know what is happening, a followed all the steps on Apple Developer webpage, a created a new Identifier, with Sign In With Apple capability allowed, a create a new Key, and associate to my identifier, and then, I went to xcode, and put my bundle identifier and selected my developer team, ok, I did the build with xcode, and they ran fine, my login with apple worked, I committed my changes, but when other developer on my team went try to run, they broke with this fucking error again,  `com.apple.AuthenticationServices.AuthorizationError` error 1000, I tried to create a new identifier and a new key, but this error persist...

**Comment by DanPHP7** [comment]
my identifiers
![img](https://i.imgur.com/20SH8vI.png)

Sign In Capability allowed
![config](https://i.imgur.com/HjAWHCa.png)

My key associated to my identifier
![key](https://i.imgur.com/reqBdPT.png)

**Comment by mikehardy** [comment]
@DanPHP7 entitlement synchronization is quite difficult on iOS builds, I only found sanity with fastlane (a reasonable handle on it here https://carloscuesta.me/blog/shipping-react-native-apps-with-fastlane#code-sign ) - if you've not done that, it could be that things aren't in sync and the entitlements aren't correct somehow. If you have done that then I dunno - seeing it work on one machine implies it's not this library, this library is just the messenger...

**Comment by DanPHP7** [comment]
Thank you @mikehardy , I'll try this.

**Comment by augustosamame** [comment]
+1 with doing this via fastlane. No issues even with multiple devs while
using it.

**Comment by eggybot** [comment]
@mikehardy I was experiencing this issue on my actual device (iPhone 7 and 8). I notice if I tap the home button to login with **short tap**, this error `com.apple.AuthenticationServices.AuthorizationError error 1000`  is showing, BUT if I **long press** till it show success... no error showing and it proceed me to login. Is this some kind of bug?

I'm using the latest version of `RN Apple Auth v1.1.2` with ` RN v0.63.2`

**Comment by mikehardy** [comment]
That sounds like a legitimate error like it just couldn't read the fingerprint

**Comment by mikehardy** [comment]
There are a lot of reasons error 1000 can be shown. Any that we have diagnosed fully end up in the troubleshooting section of the readme and we welcome PRs to add more. But I think this issue as posted is no longer an issue. Please open new issues or post PRs to the readme if there's a new way to cause 1000 errors

**Comment by oliseonwu** [comment]
**FOR EXPO** 

Hello everyone, I just wanted to share want worked for me. I simply followed the instructions [here](https://docs.expo.dev/versions/latest/sdk/apple-authentication/#eas-build) :

1. Added "usesAppleSignIn": true to my app.json
 {
  "expo": {
    "ios": {
      "usesAppleSignIn": true
    }
  }
}

2. added expo-apple-authentication to my plugins in app.json
{
  "expo": {
    "plugins": ["expo-apple-authentication"]
  }
}

3. ran "npx expo install expo-apple-authentication" inn my terminal.

NOTE: I noticed that number 2 helps us with the signin and capabilities setup automatically when we create a was build. Also number activates signin with apple.
